lib/hallmonitor/outputters/dogstatsd_outputter.rb

```diff
@@ -0,0 +1,90 @@
+begin
+  require 'datadog/statsd'
+rescue LoadError
+end
+
+module Hallmonitor
+  module Outputters
+    # An outputter for Dogstatsd
+    class DogstatsdOutputter < Outputter
+      # Builds a new DogstatsdOutputter.
+      # @param prefix [String] Prefix for all events output by this outputter,
+      #   the prefix will be applied to all event names before sending to statsd
+      # @param host [String] Datadog Host, defaults to '127.0.0.1'
+      # @param port [Number] Datadog Port, defaults to 8125
+      # @raise if Datadog::Statsd is undefined (Gem not present)
+      def initialize(prefix, host = '127.0.0.1', port = 8125, tags: {})
+        unless defined?(Datadog::Statsd)
+          fail 'In order to use DogstatsdOutputter, dogstatsd-ruby gem must be installed'
+        end
+
+        super(prefix)
+        @tags = {}.merge(tags)
+        @statsd = Datadog::Statsd.new(host).tap { |sd| sd.namespace = name }
+      end
+
+      # Sends events to statsd instance
+      # If the event's value field is a hash, this will send multiple events
+      # to statsd with the original name suffixed by the name of the events
+      # in the hash
+
+      def process_tags(tags)
+        @tags = @tags.merge(tags)
```

Comment:
This will update the `@tags` ivar over time, making every stat include all tags that have ever come through, I think you'll want to use a local var on the left side here instead of `@tags`:
`all_tags = @tags.merge(tags)`

Alternatively, you could make this a little shorter by making the whole method look like this:

`@tags.merge(tags).map {|key, value| "#{key}:#{value}"}`